### PR TITLE
Upgrade to Ruby 2.7

### PR DESCRIPTION
**Pull request by rawOrlando**
```
Psych::DisallowedClass:
        Tried to load unspecified class: Symbol
```

I ran `bundle update` which updated the packages that were not pinned. This seemed fixed some connection issues in the tests. And got farther to hit this error. If these new packages are causing the problem we should pin them. If not one of theses fixed my issue with tests that used js

```
Failure/Error:       driver.browser.file_detector = lambda do |argss|         str = argss.first.to_s         str if File.exist?(str)       end     Net::ReadTimeout:       Net::ReadTimeout with #<TCPSocket:(closed)>
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ruby-version: ['2.6']
+        ruby-version: ['2.7']
```
Comment (rawOrlando) [inline .github/workflows/ci.yml:30]: wow 👀, the image was already 2.7 so will we need to heavily test this change?

Comment (mkorcy) [reply .github/workflows/ci.yml:30]: One of the changes you made to Gemfile.lock required 2.7, so I had to bump it here.

Comment (mkorcy) [reply .github/workflows/ci.yml:30]: If I remember correctly it was the nokogiri upgrade.

**Comment by rawOrlando** [comment]
hmmmm I some how broke how the tests were running.

**Comment by rawOrlando** [comment]
Sari has looked at this in dev, and thinks it looks good.

> Without building a whole new exhibit, all major functions appear to be working as expected!

File: config/environments/development.rb [hunk 2]
```diff
@@ -8,17 +10,18 @@
 
   # Do not eager load code on boot.
   config.eager_load = false
+  config.active_record.yaml_column_permitted_classes = [Symbol, Hash, HashWithIndifferentAccess]
```
Comment (mkorcy) [inline config/environments/development.rb:13]: Can you add a comment here about why we had to do this?  I feel like in the middle of these other changes we'll look back on this confused

File: config/environments/production.rb [hunk 3]
```diff
@@ -12,6 +14,7 @@
   # and those relying on copy on write to perform better.
   # Rake tasks automatically ignore this option for performance.
   config.eager_load = true
+  config.active_record.yaml_column_permitted_classes = [Symbol, Hash, HashWithIndifferentAccess]
```
Comment (mkorcy) [inline config/environments/production.rb:17]: same comment as previous

File: config/environments/tdldev.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+Rails.application.configure do
+  # Settings specified here will take precedence over those in config/application.rb.
+
+  # In the development environment your application's code is reloaded on
+  # every request. This slows down response time but is perfect for development
+  # since you don't have to restart the web server when you make code changes.
+  config.cache_classes = false
+  config.active_record.yaml_column_permitted_classes = [Symbol, Hash, HashWithIndifferentAccess]
```
Comment (mkorcy) [inline config/environments/tdldev.rb:10]: same comment as last.

File: config/environments/test.rb [hunk 5]
```diff
@@ -11,11 +13,12 @@
   # just for the purpose of running a single test. If you are using a tool that
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
+  config.active_record.yaml_column_permitted_classes = [Symbol, Hash, HashWithIndifferentAccess]
```
Comment (mkorcy) [inline config/environments/test.rb:16]: same comment as last

File: config/initializers/translation.rb [hunk 6]
```diff
@@ -8,10 +10,10 @@
   # turn on the ActiveRecord backend, uncomment the following lines.
 
   # I18n.backend = I18n::Backend::ActiveRecord.new
-  I18n::Backend::ActiveRecord.send(:include, I18n::Backend::Memoize)
-  Translation.send(:include, Spotlight::CustomTranslationExtension)
-  I18n::Backend::Simple.send(:include, I18n::Backend::Memoize)
-  I18n::Backend::Simple.send(:include, I18n::Backend::Pluralization)
+  I18n::Backend::ActiveRecord.include I18n::Backend::Memoize
```
Comment (mkorcy) [inline config/initializers/translation.rb:13]: Whats the premise of making this change?

File: lib/tasks/test.rake [hunk 7]
```diff
@@ -32,13 +32,11 @@ namespace :tufts do
...
     %w[blacklight solr ldap].each do |f|
-      unless File.exist?("config/#{f}.yml")
-        FileUtils.cp(
-          Rails.root.join("config/#{f}.yml.sample"),
-          Rails.root.join("config/#{f}.yml")
-        )
-      end
+      next if File.exist?("config/#{f}.yml")
+      FileUtils.cp(
+        Rails.root.join("config/#{f}.yml.sample"),
```
Comment (mkorcy) [inline lib/tasks/test.rake:37]: The point of the join here is not to have the path separator in the expression, I'm guessing rubocop did this its a little weird. Rails.root.join('config',"#{f}.yml.sample") may be preferred? I'm not sure, I don't feel particularly strongly either way. probably fine to leave as is.

File: spec/features/feature_page_customizations_spec.rb [hunk 8]
```diff
@@ -30,43 +32,42 @@
...
   end
 
-  before(:each) do
+  before do
     visible_page
     sign_in(exhibit_admin)
   end
 
-  scenario "'In menu' checkbox exists in form, and sets value in db" do
+  scenario "In menu' checkbox exists in form, and sets value in db" do
```
Comment (mkorcy) [inline spec/features/feature_page_customizations_spec.rb:45]: again minor, I think 'In menu' is intentionally quoted here.

File: spec/helpers/spotlight_helper_spec.rb [hunk 9]
```diff
@@ -1,8 +1,10 @@
+# frozen_string_literal: true
+
 require 'rails_helper'
 
 RSpec.describe SpotlightHelper do
-  let(:fake_field) do { value: ['test.com'] } end
-  let(:link) { "<a target=\"_blank\" href=\"test.com\">Click to view object</a>" }
+  let(:fake_field) { { value: ['test.com'] } }
+  let(:link) { "<a target=\"_blank\" rel=\"noopener\" href=\"test.com\">Click to view object</a>" }
```
Comment (mkorcy) [inline spec/helpers/spotlight_helper_spec.rb:7]: what happened here? why the rel attribute addition?

**Comment by mkorcy** [comment]
I renamed the PR, it was driving me nuts.

**Review by mkorcy — CHANGES_REQUESTED** [review]
Only a few comments for an addition of a comment, I searched through all the code in a checked out version to try to find any potential frozen string literal issues which we've had problems with but don't see any, things looking generally good here.

File: spec/helpers/spotlight_helper_spec.rb (hunk 9 above)
Comment (rawOrlando) [reply spec/helpers/spotlight_helper_spec.rb:7]: https://www.rubydoc.info/gems/rubocop-rails/RuboCop/Cop/Rails/LinkToBlank
This was following rubicon recommendations.

File: lib/tasks/test.rake (hunk 7 above)
Comment (rawOrlando) [reply lib/tasks/test.rake:37]: rubicon here wanted to change an unless to a next if. but your point about rails join is well received

File: spec/helpers/spotlight_helper_spec.rb (hunk 9 above)
Comment (rawOrlando) [reply spec/helpers/spotlight_helper_spec.rb:7]: The change is also in [app/helpers/spotlight_helper.rb](https://github.com/TuftsUniversity/tufts-spotlight/pull/25#diff-7e456ab3da93bc5265fa5341069e9915163669724946debf6b677fde240e5488)

**Review by mkorcy — APPROVED** [review]
👍

**Pull request merged**